### PR TITLE
Integrate cargo-vet into CI system

### LINKED ISSUE #6002: Use `cargo-vet` to establish trust in Rust dependencies
`cargo-vet` is being developed by Mozilla, as a lightweight alternative to `cargo-crev`. Audits are either recorded locally, or imported from a manually-trusted foreign source. Either way, the audits are stored in the repository subject to its existing access controls, rather than relying on a separate set of keys and a web of trust, so it should be very simple to integrate.

We currently perform our Rust dependency updates for `zcashd` by hand (not using GitHub's dependabot, so integration workflow issues like https://github.com/mozilla/cargo-vet/issues/151 don't apply), so amending `cargo update` to instead be `cargo update && cargo vet` is a straightforward way for us to start building up a corpus of audited dependencies.

We currently block releases on requiring that our C++ dependencies are either up-to-date, or have recent updates postponed (via a script that @defuse wrote). Adding a `cargo-vet` check to CI might slow down the rate at which we update our Rust dependencies, but there are two things that help us here:
- @defuse's dependency-checking release script does not check Rust dependencies are up-to-date, so not updating Rust dependencies won't currently block releases.
- Just as we can postpone C++ dependency updates, we can do the inverse here and mark Rust updates as `unaudited` to allow them to pass `cargo vet`. But that functionality is intended to help with onboarding; exercising it regularly would reduce the benefit that `cargo vet` might provide us.

In any case, this is obviously safe to try, as `cargo vet` is managed as a set of text files, that we can always fix or remove if we decide to alter our approach later.

### PR DESCRIPTION
**Pull request by str4d**
Closes #6002.

### CODE REVIEW

File: qa/supply-chain/audits.toml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+
+# cargo-vet audits file
+
+[criteria.crypto-reviewed]
+description = "The cryptographic code in this crate has been reviewed for correctness by a member of a designated set of cryptography experts within the project."
+
+[criteria.license-reviewed]
+description = "The license of this crate has been reviewed for compatibility with its usage in this repository. If the crate is not available under the MIT license, `contrib/debian/copyright` has been updated with a corresponding copyright notice for files under `depends/*/vendored-sources/CRATE_NAME`."
```
Comment (str4d) [inline qa/supply-chain/audits.toml:8]: I've added this as a possibly-useful criteria, but it probably has enough overlap with `cargo-deny` that the latter might be more useful here. See also https://github.com/mozilla/cargo-vet/issues/82#issuecomment-1149660622 where `cargo-deny` might end up using `cargo-vet` as a library to offer a superset of its functionality.

File: qa/supply-chain/audits.toml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+[[audits.equihash]]
```
Comment (daira) [inline qa/supply-chain/audits.toml:9]: This probably also should have a `criteria = "crypto-reviewed"` entry.

**Review by daira — APPROVED** [review]
utACK with comment.

File: qa/supply-chain/audits.toml (hunk 2 above)
Comment (str4d) [reply qa/supply-chain/audits.toml:9]: I was undecided on whether the code in the `equihash` crate counted as cryptographic. It definitely _uses_ cryptographic code (namely BLAKE2b).

In any case, this doesn't have to block the PR, as the `crypto-reviewed` criteria is not enforced anywhere yet (I'm still figuring out precisely how policy enforcement beyond the default of `safe-to-deploy` works, but decided we should prime the audits list with the obvious things to start with).

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash/compare/395dbd3dc78cb8eb6f40c7a6976b3b0b56e9ebb2..ee256e23509324319b277ecac420037aed50b011) to try and fix the CI workflow so that all dependencies for all targets are fetched (since `cargo vet --locked` requires everything to be fetched up-front).

File: qa/supply-chain/audits.toml (hunk 2 above)
Comment (daira) [reply qa/supply-chain/audits.toml:9]: It's cryptographic in my book. But as you say, it doesn't matter much yet.

**Review by daira — COMMENTED** [review]
Post-hoc utACK

**Pull request merged**